stl/inc/format

```diff
@@ -676,6 +676,9 @@ public:
     }
 
     constexpr void _On_type(_CharT _Type) {
+        if (_Type < 0 || _Type > (numeric_limits<signed char>::max)()) {

```

Comment:
Nitpick but probably just `char`?

Reply:
the reason I used signed char is that it also eliminates the start of multi-byte characters. All the valid format specifiers are in the basic 7-bit range, and are the same in all the char formats we support.

we could go through and check that the actual char we got is one of the allowed ones, but that doesn't verify that it's allowed for the type that the format argument ends up being.

Reply:
Right, I forgot that `signed char` isn't `char`.